pkg/svcat/service-catalog/plan.go

```diff
@@ -172,6 +173,54 @@ func (sdk *SDK) RetrievePlanByClassAndName(className, planName string, opts Scop
 	return sdk.retrieveSinglePlanByListOptions(strings.Join(ss, "/"), opts, listOpts)
 }
 
+// RetrievePlanByClassIDAndName gets a plan by its external name and class kube name combination.
+func (sdk *SDK) RetrievePlanByClassIDAndName(classKubeName, planName string, scopeOpts ScopeOptions) (Plan, error) {
+	var classRefSelector fields.Selector
+	findError := &multierror.Error{
+		ErrorFormat: func(errors []error) string {
+			return joinErrors("error:", errors, "\n  ")
+		},
+	}
+
+	if scopeOpts.Scope.Matches(ClusterScope) {
+		classRefSelector = fields.OneTermEqualSelector(FieldClusterServiceClassRef, classKubeName)
+		listOpts := metav1.ListOptions{
+			FieldSelector: fields.AndSelectors(
+				classRefSelector,
+				fields.OneTermEqualSelector(FieldExternalPlanName, planName),
+			).String(),
+		}
+
+		ss := []string{classKubeName, planName}
+		plan, err := sdk.retrieveSinglePlanByListOptions(strings.Join(ss, "/"), scopeOpts, listOpts)
+		if err != nil {
+			findError = multierror.Append(findError, err)
+		}
+		if err == nil && plan != nil {
+			return plan, nil
+		}
+	}
+	if scopeOpts.Scope.Matches(NamespaceScope) {
+		classRefSelector = fields.OneTermEqualSelector(FieldServiceClassRef, classKubeName)
+		listOpts := metav1.ListOptions{
+			FieldSelector: fields.AndSelectors(
+				classRefSelector,
+				fields.OneTermEqualSelector(FieldExternalPlanName, planName),
+			).String(),
+		}
+
+		ss := []string{classKubeName, planName}
+		plan, err := sdk.retrieveSinglePlanByListOptions(strings.Join(ss, "/"), scopeOpts, listOpts)
+		if err != nil {
+			findError = multierror.Append(findError, err)
+		}
+		if err == nil && plan != nil {
```

Comment:
Do we have to check for both conditions here? I kind of expected to see this instead:

```go
if err != nil {
  findError = multierror.Append(findError, err)
  return nil, fmt.Errorf("plan '%s' not found:%s", planName, findError.Error())
}
return plan, nil
```

The convention in Go is that if the error is empty, that we should be able to trust that the plan was retrieved and populated.

Otherwise if it could be nil, then the logical fallthrough isn't handling that case properly.

Reply:
so, you're right that this can be cleaned up a bit, the problem is i can't return on err != nil, because the err could be (clusterserviceplan foo not found) when the foo I'm looking for is a namespaced service broker. This is so I can support AllScope so the cmd package (and by extension the user) doesn't have to care about figuring out on it's own whether the requested plan is cluster or namespace scoped.

I'll clean it up a bit.